integration-tests/src/lib.rs

```diff
@@ -360,15 +387,35 @@ pub fn initialize_local_test_logging(test_id: &TestId) -> Dispatch {
             .with_filter(filter.parse::<EnvFilter>().unwrap())
     });
 
-    subscriber.with(stdout_log).with(file_log).into()
+    let (flame_log, guard) = LOG_FLAME_ENV_FILTER
+        .as_ref()
+        .map(|(filter, _)| {
+            let path = test_id.make_artifact_file_path("tracing.folded").unwrap();
+            eprintln!("Flamegraph data written to: {}", path.display());
+            let (layer, guard) = FlameLayer::with_file(path).unwrap();
+
+            (
+                layer.with_filter(filter.parse::<EnvFilter>().unwrap()),
+                guard,
+            )
+        })
+        .unzip();
+
+    let dispatch = subscriber
+        .with(stdout_log)
+        .with(file_log)
+        .with(flame_log)
+        .into();
+
+    (dispatch, guard)
 }
 
 pub fn run_async_test<F>(test: impl FnOnce(TestId) -> F)
 where
     F: Future<Output = Result<(), Error>>,
 {
     let test_id = TestId::generate();
-    let subscriber = initialize_local_test_logging(&test_id);
+    let (subscriber, _guard) = initialize_local_test_logging(&test_id);
```

Comment:
❓ similar question to the previous pr: don't we have to explicitely `drop(guard)` at the end of the function to prevent that it might get dropped already in the next line?

Reply:
`_guard` gets dropped at the end of the scope, but I would love to add a manual `drop` for more clearity